src/services/document-service.js

```diff
@@ -255,9 +262,17 @@ class DocumentService {
     try {
       lock = await this.lockStore.takeDocumentLock(documentId);
 
-      logger.info(`Hard deleting external document '${documentId}'`);
+      logger.info(`Hard deleting document '${documentId}'`);
 
       await this.transactionRunner.run(async session => {
+        const doc = await this.documentStore.getDocumentById(documentId, { session });
+
+        if (doc.roomId) {
```

Comment:
I'd also lock the room for this operation

Reply:
Done here too.
And changed the logic for updating rooms as we discussed 🙏

Reply:
Nice! Looks good, thanks!